XiEditor/CoreConnection.swift

```diff
@@ -66,23 +79,72 @@ class CoreConnection {
         } else {
             self.rpcLogWriter = nil
         }
-        let task = Process()
+        
+        let tmpErrLog = logDirectory.appendingPathComponent("xi_tmp.log").path
+        
+        self.errLogWriter = FileWriter(path: tmpErrLog)
+        if self.errLogWriter != nil {
+            print("logging stderr to \(tmpErrLog)")
+        }
+          
         task.launchPath = path
         task.arguments = []
         let outPipe = Pipe()
         task.standardOutput = outPipe
         let inPipe = Pipe()
         task.standardInput = inPipe
+        let errPipe = Pipe()
+        task.standardError = errPipe
         inHandle = inPipe.fileHandleForWriting
         recvBuf = Data()
-
+        
         outPipe.fileHandleForReading.readabilityHandler = { handle in
             let data = handle.availableData
             self.recvHandler(data)
         }
+        
+        errPipe.fileHandleForReading.readabilityHandler = { handle in
+            let data = handle.availableData
+            var lineCount = 1
+            
+            if let errString = String(data: data, encoding: String.Encoding.utf8) {
+                print(errString, terminator: "")
+                self.errOutput += errString
+            }
+            
+            if self.errOutput.hasSuffix("\n") {
+                lineCount += 1
+                if lineCount >= 100 {
+                    self.errOutput = ""
+                    lineCount = 1
+                }
+            }
+        }
+        
+        // write to log on xi-core crash
+        task.terminationHandler = { _ in
+            // get current date to use as timestamp
+            let dateFormatter = DateFormatter()
+            let currentTime = Date.init()
+            dateFormatter.dateFormat = "yyyy-MM-dd-HHMMSS"
+            let timeStamp = dateFormatter.string(from: currentTime)
+            
+            let tmpErrLog = self.logDirectory.appendingPathComponent("xi_tmp.log")
```

Comment:
see earlier comment about having this be declared only once

Reply:
+1.